2.0/gradle/vividus-test.gradle

```diff
@@ -158,6 +158,63 @@ tasks.withType(StoriesExec).configureEach {
 
 ext.VIVIDUS_TASK_GROUP = 'VIVIDUS'
 
+tasks.register('checkDependenciesConsistency') {
+    dependsOn = [test]
+    group = VIVIDUS_TASK_GROUP
+    description = 'Ensures consistency of VIVIDUS dependencies in the project.'
+
+    doFirst {
+        def checkConsistency = project.findProperty('vividus.check-dependencies-consistency')
+        if (checkConsistency == null || !checkConsistency.toBoolean()) {
+            return
+        }
+
+        def dependencies = project.configurations.implementation.allDependencies;
+        def groupMatcher = { it.group.equals('org.vividus') }
+        def nameMatcher = { dependency, targetName ->
+            groupMatcher.call(dependency) && dependency.name.equals(targetName)
+        }
+
+        def vividusLib = dependencies.find { nameMatcher.call(it, 'vividus') }
+        if (vividusLib == null) {
+            throw new GradleException('The mandatory \'vividus\' dependency is missing in project dependencies'
+                + ' configuration.')
+        }
+
+        def vividusBomLib = dependencies.find { nameMatcher.call(it, 'vividus-bom') }
+        if (vividusBomLib != null) {
+            def vividusBomVersion = vividusBomLib.version
+            if (vividusBomVersion == null) {
+                throw new GradleException('Please specify version for \'vividus-bom\' dependency in project'
```

Comment:
won't it fail earlier?

Reply:
suprisingly its not, only for impementation dependencies

Reply:
discussed